### PR TITLE
[storage] fix StateStore latest version fetching wrt PRE_GENESIS      

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

make sure it works for PRE_GENESIS, until we decide to remove the support

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?
Y
## Test Plan
unit tests

### CODE REVIEW

**Review by lightmark — APPROVED** [review]

File: storage/aptosdb/src/state_store/mod.rs [hunk 1]
```diff
@@ -72,31 +72,38 @@ impl StateStore {
...
+            let mut iter = db.rev_iter::<JellyfishMerkleNodeSchema>(Default::default())?;
+            iter.seek_for_prev(&NodeKey::new_empty_path(max_possible_version))?;
+            if let Some((key, _node)) = iter.next().transpose()? {
+                // TODO: If we break up a single update batch to multiple commits, we would need to
+                // deal with a partial version, which hasn't got the root committed.
+                return Ok(Some(key.version()));
+            }
+        }
+        // try PRE_GENESIS
+        Ok(db
```
Comment (sitalkedia) [inline storage/aptosdb/src/state_store/mod.rs:104]: nice find and fix. Curious to know how you found this issue?

**Review by sitalkedia — APPROVED** [review]

File: storage/aptosdb/src/state_store/mod.rs (hunk 1 above)
Comment (msmouse) [reply storage/aptosdb/src/state_store/mod.rs:104]: When you code shitty code, you know there's a bug. :trollface:

**Pull request closed without merging**